link.go

```diff
@@ -512,10 +512,21 @@ func (l *link) muxReceive(fr performTransfer) error {
 		return err
 	}
 	debug(1, "deliveryID %d before push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", l.msg.deliveryID, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
-	// send to receiver, this should never block due to buffering
-	// and flow control.
+
 	l.addUnsettled(&l.msg)
-	l.messages <- l.msg
+
+	// send to receiver, this should never block due to buffering and flow
+	// control, but we still check for link close in case something unexpected
+	// happens so that we can shut down properly.
+	var closed bool
+	select {
+	case l.messages <- l.msg:
+		// Sent. No-op
+	case <-l.close:
+		// Link was closed
+		debug(1, "deliveryID %d skipping push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", l.msg.deliveryID, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
+		closed = true
+	}
 
 	debug(1, "deliveryID %d after push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", l.msg.deliveryID, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
```

Comment:
Should this debug line move into the select?

Reply:
good catch! moved